apps/meteor/app/livechat/server/lib/LivechatTyped.ts

```diff
@@ -459,6 +460,35 @@ class LivechatClass {
 			extraData,
 		});
 
+		if (isSingleContactEnabled()) {
+			let { contactId } = visitor;
+			if (contactId) {
+				const visitorContact = await LivechatVisitors.findOne<Pick<ILivechatVisitor, 'contactId'>>(visitor._id, {
+					projection: { contactId: 1 },
+				});
+				contactId = visitorContact?.contactId;
+			}
+
+			const contact = await LivechatContacts.findOne({ _id: contactId });
```

Comment:
I think models have a `findOneById` by default

Reply:
Also, do you need the full `contact` object? Or can you go with just the _id & channels prop?